test/features/push_notification/presentation/controller/push_base_controller_test.dart

```diff
@@ -0,0 +1,230 @@
+          emailChangeListener: emailChangeListener,
+          mailboxChangeListener: mailboxChangeListener
+        );
+
+        // assert
+        verify(
+          emailChangeListener.dispatchActions([
+            StoreEmailStateToRefreshAction(
+              TypeName.emailType,
+              state,
+              accountId,
+              userName,
+              null,
+            ),
+          ]),
+        ).called(1);
+      });
+
+      test(
+        'should call emailChangeListener.dispatchActions with nothing '
+        'when mapTypeState contains emailDelivery '
+        'and isForeground is true',
+      () {
+        // arrange
+        final state = State('some-state');
+        final mapTypeState = {TypeName.emailDelivery.value: state.value};
+
+        // act
+        final pushBaseController = TestPushController();
+        pushBaseController.mappingTypeStateToAction(
+          mapTypeState,
+          accountId,
+          userName,
+          isForeground: true,
+          emailChangeListener: emailChangeListener,
+          mailboxChangeListener: mailboxChangeListener
+        );
+
+        // assert
+        verifyNever(emailChangeListener.dispatchActions(any));
+      });
+
+      test(
+        'should call emailChangeListener.dispatchActions with PushNotificationAction '
+        'when mapTypeState contains emailDelivery '
+        'and isForeground is false',
+      () {
+        // arrange
+        final state = State('some-state');
+        final mapTypeState = {TypeName.emailDelivery.value: state.value};
+
+        // act
+        final pushBaseController = TestPushController();
+        pushBaseController.mappingTypeStateToAction(
+          mapTypeState,
+          accountId,
+          userName,
+          isForeground: false,
+          emailChangeListener: emailChangeListener,
+          mailboxChangeListener: mailboxChangeListener
+        );
+
+        // assert
+        verify(
+          emailChangeListener.dispatchActions([
+            PushNotificationAction(
+              TypeName.emailDelivery,
+              state,
+              null,
+              accountId,
+              userName,
+            ),
+          ]),
+        ).called(1);
+      });
+      
+      test(
+        'should call mailboxChangeListener.dispatchActions with SynchronizeMailboxOnForegroundAction '
+        'when mapTypeState contains mailboxType '
+        'and isForeground is true',
+      () {
+        // arrange
+        final state = State('some-state');
+        final mapTypeState = {TypeName.mailboxType.value: state.value};
+
+        // act
+        final pushBaseController = TestPushController();
+        pushBaseController.mappingTypeStateToAction(
+          mapTypeState,
+          accountId,
+          userName,
+          isForeground: true,
+          emailChangeListener: emailChangeListener,
+          mailboxChangeListener: mailboxChangeListener
+        );
+
+        // assert
+        verify(
```

Comment:
should make sure `emailChangeListener` should not be invoked

Reply:
Done